File: day1.py


file = open("adventofcode2021/day1.txt", "r")
depth = file.read().split("\n")



def starOne():
    counter = 1

    for i in range(len(depth) - 1):
        if depth[i] < depth[i+1]:
            counter += 1
    return counter



def starTwo():
    counter = 0
    for i in range(len(depth) - 3):
        if (int(depth[i]) + int(depth[i+1]) + int(depth[i+2])) < (int(depth[i+1]) + int(depth[i+2]) + int(depth[i+3])):
            counter += 1
    return counter

print("Star 1: ", starOne())
print("Star 2: ", starTwo())

File: day1.cs
using System;
using System.IO;
using System.Text;

namespace Übungen
{
    class MainClass
    {
        public static void Main(string[] args)
        {
            string text = File.ReadAllText("../../Input.txt");
            string[] sNumbers = text.Split('\n');

            Console.WriteLine($"Star 1: {StarOne(sNumbers)}");
            Console.WriteLine($"Star 2: {StarTwo(sNumbers)}");
        }


        public static int StarOne(string[] numbers)
        {
            int counter = 0;
            int comp = 0;

            foreach (string value in numbers)
            {
                if (comp != 0 && comp < System.Convert.ToInt32(value))
                {
                    counter++;
                }
                comp = System.Convert.ToInt32(value);
            }

            return counter;
        }


        public static int StarTwo(string[] numbers)
        {
            int counter = 0;

            for (int i = 0; i < numbers.Length - 3; i++)
            {
                if (System.Convert.ToInt32(numbers[i]) + System.Convert.ToInt32(numbers[i+1]) + System.Convert.ToInt32(numbers[i+2]) < System.Convert.ToInt32(numbers[i+1]) + System.Convert.ToInt32(numbers[i+2]) + System.Convert.ToInt32(numbers[i+3]))
                {
                    counter++;
                }
            }

            return counter;
        }
    }
}
